src/Sylius/Bundle/ApiBundle/EventListener/DuplicateEntryExceptionListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventListener;
+
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpFoundation\Response;
+
+class DuplicateEntryExceptionListener
```

Comment:
This problem should be handled by validation instead of the event listener. If the issue is connected with taxons, there is probably something missing that skips entity validation, where uniqueness should be checked